Framework/PythonInterface/plugins/algorithms/HB3AIntegrateDetectorPeaks.py

```diff
@@ -0,0 +1,203 @@
+        self.declareProperty("SignalNoiseMin", 1.0,
+                             doc="Minimum Signal/Noice ratio (Intensity/SigmaIntensity) of peak to be added to the output")
+        self.declareProperty("ApplyLorentz", True, doc="If to apply Lorentz Correction to intensity")
+
+        self.declareProperty("OutputFitResults", False, doc="This will output the fitting result workspace and a ROI workspace")
+
+        self.declareProperty("OptimizeQVector", True,
+                             doc="This will convert the data to q and optimize the peak location using CentroidPeaksdMD")
+
+        self.declareProperty(IPeaksWorkspaceProperty("OutputWorkspace", "", optional=PropertyMode.Mandatory, direction=Direction.Output),
+                             doc="Output Peaks Workspace")
+
+    def PyExec(self):
+        input_workspaces = self._expand_groups()
+        outWS = self.getPropertyValue("OutputWorkspace")
+        CreatePeaksWorkspace(OutputType='LeanElasticPeak',
+                             InstrumentWorkspace=input_workspaces[0],
+                             NumberOfPeaks=0,
+                             OutputWorkspace=outWS, EnableLogging=False)
+
+        scale = self.getProperty("ScaleFactor").value
+        chisqmax = self.getProperty("ChiSqMax").value
+        signalNoiseMin = self.getProperty("SignalNoiseMin").value
+        ll = self.getProperty("LowerLeft").value
+        ur = self.getProperty("UpperRight").value
+        startX = self.getProperty('StartX').value
+        endX = self.getProperty('EndX').value
+        use_lorentz = self.getProperty("ApplyLorentz").value
+        optmize_q = self.getProperty("OptimizeQVector").value
+        output_fit = self.getProperty("OutputFitResults").value
+
+        if output_fit:
+            fit_results = WorkspaceGroup()
+            AnalysisDataService.addOrReplace(outWS+"_fit_results", fit_results)
+
+        for inWS in input_workspaces:
+            tmp_inWS = '__tmp_' + inWS
+            IntegrateMDHistoWorkspace(InputWorkspace=inWS,
+                                      P1Bin=f'{ll[1]},{ur[1]}',
+                                      P2Bin=f'{ll[0]},{ur[0]}',
+                                      OutputWorkspace=tmp_inWS,
+                                      EnableLogging=False)
+            ConvertMDHistoToMatrixWorkspace(tmp_inWS, OutputWorkspace=tmp_inWS, EnableLogging=False)
+            data = ConvertToPointData(tmp_inWS, OutputWorkspace=tmp_inWS, EnableLogging=False)
+
+            run = mtd[inWS].getExperimentInfo(0).run()
+            scan_log = 'omega' if np.isclose(run.getTimeAveragedStd('phi'), 0.0) else 'phi'
+            scan_axis = run[scan_log].value
+            data.setX(0, scan_axis)
+
+            y = data.extractY().flatten()
+            x = data.extractX().flatten()
+            function = f"name=FlatBackground, A0={y.min()}; name=Gaussian, PeakCentre={x[y.argmax()]}, Height={y.max()-y.min()}, Sigma=0.25"
```

Comment:
I think issues with NaNs can be avoided if using `np.nanmin()` ,`np.nanargmin()` ,`np.nanmax()` , and `np.nanargmax()` 

```
Invalid value for property Function (Function) from string "name=FlatBackground, A0=nan; name=Gaussian, PeakCentre=15.2605, Height=nan, Sigma=0.25": Trying to set a NaN value (nan) to parameter A0
```

Reply:
```python
function = f"name=FlatBackground, A0={np.nanmin(y)}; name=Gaussian, PeakCentre={x[np.nanargmax(y)]}, Height={np.nanmax(y)-np.nanmin(y)}, Sigma=0.25"
constraints = f"f0.A0 > 0, f1.Height > 0, {np.nanmin(x)} < f1.PeakCentre < {np.nanmax(x)}"
```

Reply:
That won't fix it because the `Fit` algorithm will also fail with `Infinte number or NaN found in input data.`

The issue is arising because the last measure point in scan 36 has time=0 and monitor=0 so any normalisation will create infinite.

If you do
```python
data = HB3AAdjustSampleNorm(Filename=data_files, OutputType='Detector', NormaliseBy='None')
```
it should work.

But I think we still need to handle the where time=0 and you want to normalise the data, I will need to think of the best way

Reply:
It seems to work by adding `IgnoreInvalidData=True` to Fit along with your changes.

Reply:
@zjmorgan Should be good now if you want to try again.

Reply:
This looks good